src/edit/contributorFields.js

```diff
@@ -74,6 +68,35 @@ const ContributorFields = ({
               name: 'contributorTypeText',
               component: TextField,
             },
+            {
+              name: 'primary',
+              label: intl.formatMessage({ id: 'ui-inventory.primary' }),
+              component: ({ label, meta, input, fields }) => { /* eslint-disable-line react/prop-types */
+                const isPrimary = input.value === true;
+                const handleChange = () => {
+                  // Reset other primary fields
+                  fields.forEach(fieldName => meta.dispatch(change(meta.form, `${fieldName}.primary`, false)));
+
+                  // Set primary flag for current field
+                  input.onChange(true);
+                };
+
+                return (
+                  <Fragment>
+                    { label && <Label>{label}</Label>}
+                    <Button
+                      data-test-edit-instance-contributors-primary-button
+                      buttonStyle={isPrimary ? 'primary' : 'default'}
+                      onClick={!isPrimary ? handleChange : null}
```

Comment:
Is it possible to move `!isPrimary ? handleChange : null` to a handler method?

Reply:
Sure 👍🏻